cluster/src/main/java/com/hartwig/pipeline/tertiary/rose/Rose.java

```diff
@@ -0,0 +1,170 @@
+import com.hartwig.pipeline.storage.GoogleStorageLocation;
+import com.hartwig.pipeline.storage.RuntimeBucket;
+import com.hartwig.pipeline.tertiary.chord.ChordOutput;
+import com.hartwig.pipeline.tertiary.cuppa.CuppaOutput;
+import com.hartwig.pipeline.tertiary.linx.LinxSomaticOutput;
+import com.hartwig.pipeline.tertiary.linx.LinxSomaticOutputLocations;
+import com.hartwig.pipeline.tertiary.purple.PurpleOutput;
+import com.hartwig.pipeline.tertiary.purple.PurpleOutputLocations;
+import com.hartwig.pipeline.tertiary.virus.VirusOutput;
+import com.hartwig.pipeline.tools.Versions;
+
+@Namespace(Rose.NAMESPACE)
+public class Rose implements Stage<RoseOutput, SomaticRunMetadata> {
+    public static final String NAMESPACE = "rose";
+    private final ResourceFiles resourceFiles;
+    private final InputDownload purplePurity;
+    private final InputDownload purpleQc;
+    private final InputDownload purpleGeneCopyNumber;
+    private final InputDownload purpleSomaticDriverCatalog;
+    private final InputDownload purpleGermlineDriverCatalog;
+    private final InputDownload purpleSomaticVariants;
+    private final InputDownload purpleGermlineVariants;
+    private final InputDownload linxFusions;
+    private final InputDownload linxBreakends;
+    private final InputDownload linxDriverCatalog;
+    private final InputDownload virusAnnotations;
+    private final InputDownload chordPredictions;
+    private final InputDownload molecularTissueOrigins;
+
+    public Rose(final ResourceFiles resourceFiles, final PurpleOutput purpleOutput, final LinxSomaticOutput linxSomaticOutput,
+            final VirusOutput virusOutput, final ChordOutput chordOutput, final CuppaOutput cuppaOutput) {
+        this.resourceFiles = resourceFiles;
+        PurpleOutputLocations purple = purpleOutput.outputLocations();
+        this.purplePurity = new InputDownload(purple.purity());
+        this.purpleQc = new InputDownload(purple.qcFile());
+        this.purpleGeneCopyNumber = initialiseOptionalLocation(purple.geneCopyNumber());
+        this.purpleSomaticDriverCatalog = initialiseOptionalLocation(purple.somaticDriverCatalog());
+        this.purpleGermlineDriverCatalog = initialiseOptionalLocation(purple.germlineDriverCatalog());
+        this.purpleSomaticVariants = initialiseOptionalLocation(purple.somaticVariants());
+        this.purpleGermlineVariants = initialiseOptionalLocation(purple.germlineVariants());
+        LinxSomaticOutputLocations linx = linxSomaticOutput.linxOutputLocations();
+        this.linxFusions = new InputDownload(linx.fusions());
+        this.linxBreakends = new InputDownload(linx.breakends());
+        this.linxDriverCatalog = new InputDownload(linx.driverCatalog());
+        this.virusAnnotations = new InputDownload(virusOutput.virusAnnotations());
+        this.chordPredictions = new InputDownload(chordOutput.predictions());
+        this.molecularTissueOrigins = new InputDownload(cuppaOutput.cuppaOutputLocations().conclusionTxt());
```

Comment:
the naming differences are a little confusing. maybe we can keep our code consistent and only the rose arg differs